rfd/0035-desktop-access-windows-authn.md

```diff
@@ -0,0 +1,119 @@
+---
+authors: Andrew Lytvynov (andrew@goteleport.com)
+state: draft
+---
+
+# RFD 35 - Desktop Access - Windows Certificate Authentication
+
+## What
+
+RFD 33 defines the high-level goals and architecture for Teleport Desktop
+Access. RFD 34 defines the integration details for Windows hosts.
+
+Desktop Access for Windows supports:
+- certificate-based authentication, by emulating smart cards
+- username/password authentication, as a fallback when certificates fail
+
+This RFD describes the details of the former, certificate-based approach.
+
+## Details
+
+### Background
+
+While Windows does not support generic mTLS or similar PKI-based
+authentication, it does support smart cards. Smart cards are similar to HSMs or
+U2F/WebAuthn tokens - they store private keys in and expose certificates and
+basic asymmetric crypto operations.
+
+The server side (Windows host or domain controller) validate the credentials
+against a trusted Certificate Authority. More information at
+https://docs.microsoft.com/en-us/troubleshoot/windows-server/windows-security/enabling-smart-card-logon-third-party-certification-authorities
+
+Smart cards are typically hardware devices, but a driver can be
+implemented to emulate them in software
+([example](http://frankmorgner.github.io/vsmartcard/virtualsmartcard/README.html)).
+Since this is the only known way for us to support cert-based authn, Teleport
+will implement a smart card emulator to authenticate over RDP. This emulator
+will use Teleport TLS certs and expose the User TLS CA for import into the
+Windows trust store.
```

Comment:
Would this be a proper Windows device driver? If so, I wonder if we could alternatively implement our smart card "driver" directly inside the RDP client. Though I'm not sure what's better: a forked RDP library or a custom Windows driver...

Reply:
this will be an emulated driver implemented in the RDP client, since the client (`windows_desktop_service`) will likely run on a Linux machine.
RDP server should establish a virtual channel back to the client to send driver IOCTLs

Reply:
Ah, makes sense, thanks!